vespabase/src/start-cbinaries.sh

```diff
@@ -152,7 +152,9 @@ configure_vespa_malloc () {
             tryfile="${VESPA_HOME}/${pre}/${suf}"
             if [ -f "$tryfile" ]; then
                 LD_PRELOAD="$tryfile"
-                log_debug_message "Using LD_PRELOAD='$tryfile'"
+                if [ "$VESPA_LOAD_CODE_AS_HUGEPAGES" ]; then
+                    LD_PRELOAD="$LD_PRELOAD:"${VESPA_HOME}/${pre}/vespa/malloc/libvespa_load_as_huge.so"
```

Comment:
Remapping of code segments might happen too early when `libvespa_load_as_huge.so` is preloaded.

Reply:
I have verified that it works as intended using LD_PRELOAD by manually editing startup script and running unit tests.